language/stdlib/modules/Libra.move

```diff
@@ -1158,73 +1126,177 @@ module Libra {
-
-        /// The permission "UpdateExchangeRate(type)" is granted to TreasuryCompliance [B14].
-        apply Roles::AbortsIfNotTreasuryCompliance{account: tc_account} to update_lbr_exchange_rate<FromCoinType>;
-    }
+    /// ## Minting
 
     spec schema TotalValueNotIncrease<CoinType> {
         /// The total amount of currency does not increase.
         ensures old(spec_is_currency<CoinType>())
             ==> spec_currency_info<CoinType>().total_value <= old(spec_currency_info<CoinType>().total_value);
     }
+    spec schema PreserveMintCapExistence<CoinType> {
+        /// The existence of MintCapability is preserved.
+        ensures forall addr: address:
+            old(exists<MintCapability<CoinType>>(addr)) ==>
+                exists<MintCapability<CoinType>>(addr);
+    }
+    spec schema PreserveMintCapAbsence<CoinType> {
+        /// The absence of MintCapability is preserved.
+        ensures forall addr: address:
+            old(!exists<MintCapability<CoinType>>(addr)) ==>
+                !exists<MintCapability<CoinType>>(addr);
+    }
     spec module {
         /// Only mint functions can increase the total amount of currency [B11].
         apply TotalValueNotIncrease<CoinType> to *<CoinType>
             except mint<CoinType>, mint_with_capability<CoinType>;
+
+        /// In order to successfully call `mint` and `mint_with_capability`, MintCapability is
+        /// required. MintCapability must be only granted to a TreasuryCompliance account [B11].
+        /// Only `register_SCS_currency` creates MintCapability, which must abort if the account
+        /// does not have the TreasuryCompliance role [B17].
+        apply PreserveMintCapAbsence<CoinType> to *<CoinType> except register_SCS_currency<CoinType>;
+        apply Roles::AbortsIfNotTreasuryCompliance{account: tc_account} to register_SCS_currency<CoinType>;
+
+        /// Only TreasuryCompliance can have MintCapability [B11].
+        /// If an account has MintCapability, it is a TreasuryCompliance account.
+        invariant [global] forall coin_type: type:
+            forall addr: address:
+                exists<MintCapability<coin_type>>(addr) ==>
+                    Roles::spec_has_treasury_compliance_role_addr(addr);
+
+        /// MintCapability is not transferrable [D11].
+        apply PreserveMintCapExistence<CoinType> to *<CoinType>;
+
+        /// The permission "MintCurrency" is unique per currency [C11].
+        /// At most one address has a mint capability for SCS CoinType
+        invariant [global, isolated]
+            forall coin_type: type where spec_is_SCS_currency<coin_type>():
+                forall addr1: address, addr2: address
+                     where exists<MintCapability<coin_type>>(addr1) && exists<MintCapability<coin_type>>(addr2):
+                          addr1 == addr2;
+        /// If an address has a mint capability, it is an SCS currency.
+        invariant [global]
```

Comment:
Need to include modules that have this as a dependency to make sure this holds?  See the comment above about "register_currency"

Reply:
The same as above.